### PR TITLE
this commit completes ticket_8 including code, testing, error handlin…

### PR DESCRIPTION
**Pull request by cheesenbeans**
…g, new util, util testing and updating endpointsjson

### CODE REVIEW

File: app.test.js [hunk 1]
```diff
@@ -89,6 +88,52 @@ describe("/api/articles/:article_id", () => {
...
+      .expect(400)
+      .send({
+        inc_votes: 10,
+      })
+      .then((result) => {
+        const article = result.body;
+        expect(article.msg).toBe("Bad Request");
+      });
+  });
+  test("PATCH request - status code 404 - invalid article id", () => {
```
Comment (cjplatten) [inline app.test.js:125]: the id is not invalid, it just doesn't exist

File: app.test.js [hunk 2]
```diff
@@ -89,6 +88,52 @@ describe("/api/articles/:article_id", () => {
...
+      .patch(`/api/articles/23141`)
+      .expect(404)
+      .send({
+        inc_votes: 10,
+      })
+      .then((result) => {
+        const article = result.body;
+        expect(article.msg).toBe("Not Found!");
+      });
+  });
```
Comment (cjplatten) [inline app.test.js:136]: what if the value submitted on the key of inc_votes is not a number?

File: models/articles.models.js [hunk 3]
```diff
@@ -50,3 +51,18 @@ exports.getCommentsByArticle = (articleId) => {
     return result.rows;
   });
 };
+
+exports.patchVotes = (articleId, votes) => {
+  return getVotes(articleId)
```
Comment (cjplatten) [inline models/articles.models.js]: think about what the purpose of this function is, is it to check if the article exists and/or to find the current votes value? 

Checking the article exists could be acheived through a more generic function and the db.query has access to to the current votes so these can both be achevied in other ways.

It is up to you if you want to make modifications based on this.

File: models/articles.models.js [hunk 4]
```diff
@@ -50,3 +51,18 @@ exports.getCommentsByArticle = (articleId) => {
...
+  .then((currentVotes) => {
+    const queryStr = `
+      UPDATE articles
+      SET votes = $1
```
Comment (cjplatten) [inline models/articles.models.js:110]: this can also be acheived with votes + $1 ($1 being the vote increment)

**Review by cjplatten — COMMENTED** [review]
good start, please see comments

**Review by HannahBurks — APPROVED** [review]
Great Chris - merge away :)

**Pull request merged**